arango/batch.py

```diff
@@ -214,6 +247,32 @@ def graph(self, name):
         return Graph(self, name)
 
 
+if six.PY2:
+    class Python2Lock(object):
+        # https://stackoverflow.com/questions/35149889/lock-with-timeout-in-python2-7
+        def __init__(self):
+            self._underlying = Queue.Queue(maxsize=1)
+            self._underlying.put(True, block=False)
+
+        def acquire(self, timeout=0):
+            if timeout > 0:
+                try:
+                    return self._underlying.get(block=True, timeout=timeout)
+                except Queue.Empty:
+                    return False
+            else:
```

Comment:
It is a common practice avoid using `else`. I personally think it's more readable that way.

Also, the default value of `timeout` would be better as `None` since we never use the value 0. It is less misleading (makes it clear to the users that they must set it to put it to use), and stays more consistent with how the rest of the library treats default values.

So I would change this method like this:

```python
def acquire(self, timeout=None):
    if not timeout:
        return self._underlying.get(block=True)
    try:
        return self._underlying.get(block=True, timeout=timeout)
    except Queue.Empty:
        return False
```
Or this would be even better if it's correct:
```python
def acquire(self, timeout=None):
    try:
        return self._underlying.get(block=True, timeout=timeout)
    except Queue.Empty:
        return False
```

Reply:
Although python has duck typing, I dislike using if statements to check "falsy" objects.  That being said, your second suggestion would be correct since the default value of timeout is None